### PR TITLE
Add support for `COUNT`

### PR DESCRIPTION
**Pull request by staab**
Based on https://github.com/nostr-protocol/nips/pull/144/files

I am having a weird compilation error, which I can fix by removing the variable declarations that are causing problems, but @fiatjaf do you have any ideas?

```
✘ [ERROR] The symbol "id" has already been declared

    relay.ts:162:18:
      162 │               let id = data[1]
          ╵                   ~~

  The symbol "id" was originally declared here:

    relay.ts:149:18:
      149 │               let id = data[1]
          ╵                   ~~

✘ [ERROR] The symbol "id" has already been declared

    relay.ts:162:18:
      162 │               let id = data[1]
          ╵                   ~~

  The symbol "id" was originally declared here:

    relay.ts:149:18:
      149 │               let id = data[1]
          ╵                   ~~

✘ [ERROR] The symbol "id" has already been declared

    relay.ts:162:18:
      162 │               let id = data[1]
          ╵                   ~~

  The symbol "id" was originally declared here:

    relay.ts:149:18:
      149 │               let id = data[1]
          ╵                   ~~

/Users/jstaab/my/nostr/nostr-tools/node_modules/esbuild/lib/main.js:1594
  let error = new Error(`${text}${summary}`);
              ^

Error: Build failed with 1 error:
relay.ts:162:18: ERROR: The symbol "id" has already been declared
    at failureErrorWithLog (/Users/jstaab/my/nostr/nostr-tools/node_modules/esbuild/lib/main.js:1594:15)
    at /Users/jstaab/my/nostr/nostr-tools/node_modules/esbuild/lib/main.js:1050:28
    at /Users/jstaab/my/nostr/nostr-tools/node_modules/esbuild/lib/main.js:995:67
    at buildResponseToResult (/Users/jstaab/my/nostr/nostr-tools/node_modules/esbuild/lib/main.js:1048:7)
    at /Users/jstaab/my/nostr/nostr-tools/node_modules/esbuild/lib/main.js:1160:14
    at responseCallbacks.<computed> (/Users/jstaab/my/nostr/nostr-tools/node_modules/esbuild/lib/main.js:697:9)
    at handleIncomingPacket (/Users/jstaab/my/nostr/nostr-tools/node_modules/esbuild/lib/main.js:752:9)
    at Socket.readFromStdout (/Users/jstaab/my/nostr/nostr-tools/node_modules/esbuild/lib/main.js:673:7)
    at Socket.emit (node:events:527:28)
    at addChunk (node:internal/streams/readable:315:12) {
  errors: [
    {
      detail: undefined,
      id: '',
      location: {
        column: 18,
        file: 'relay.ts',
        length: 2,
        line: 162,
        lineText: '              let id = data[1]',
        namespace: '',
        suggestion: ''
      },
      notes: [
        {
          location: {
            column: 18,
            file: 'relay.ts',
            length: 2,
            line: 149,
            lineText: '              let id = data[1]',
            namespace: '',
            suggestion: ''
          },
          text: 'The symbol "id" was originally declared here:'
        }
      ],
      pluginName: '',
      text: 'The symbol "id" has already been declared'
    }
  ],
  warnings: []
}
```

### CODE REVIEW

**Review by adamritter — COMMENTED** [review]
An option to think about is to pass array of filters instead of 1 filter to count, as it will be hard to modify the API later.

An example is getting number of followers for all followed authors (there are other examples written by Giszmo in the NIP where it may be useful).

As the server returns all counts in 1 message, it's still 1 promise, so I don't see much downside in the generalization.

(also probably you wanted Promise<CountPayload | null> or Promise<CountPayload[].| null> in the return value for count

**Comment by staab** [comment]
Good catches, updated

File: relay.ts [hunk 1]
```diff
@@ -22,6 +26,7 @@ export type Relay = {
   sub: (filters: Filter[], opts?: SubscriptionOptions) => Sub
   list: (filters: Filter[], opts?: SubscriptionOptions) => Promise<Event[]>
   get: (filter: Filter, opts?: SubscriptionOptions) => Promise<Event | null>
+  count: (filter: Filter[], opts?: SubscriptionOptions) => Promise<CountPayload | null>
```
Comment (adamritter) [inline relay.ts:29]: Name should be filters for array

**Comment by verbiricha** [comment]
lgtm

**Pull request merged**